andela/Ah-backend-xmen
# #162725267 Enable users to add tags to their articles

#### What does this PR do?
This PR enables a user to add tags to his articles and view all the available tags

#### Description of Task to be completed?
- When a user has created an article, it is always good culture to give a hint what that article is all about, most especially if the article is long and someone cannot read all of it or when there are too many stories and a user wants to find stories related to a particular thing. this is where tags come in.
- should have the endpoint `/api/tags` working
- create a tag field that will handle the data from the user as tags
- create a view function that fetches all tags

#### How should this be manually tested?
- Fetch this branch by running the command `git fetch origin ft-tag-articles-162725267` in the terminal and check into the branch `git checkout ft-tag-articles-162725267`
- Start the virtual environment `source venv/bin/activate`
- Migrate so that the fields required for the data are initiated by typing this command: `python manage.py migrate`
- Start the application by running this command: `python manage.py runserver`
- Go to postman and in it create an article with a tag 

#### Screenshots
`POST  /api/articles`
<img width="1211" alt="screen shot 2019-01-28 at 11 58 45 am" src="https://user-images.githubusercontent.com/34310682/51824748-3058c780-22f4-11e9-95ac-06b8d4557bd0.png">

<img width="1211" alt="screen shot 2019-01-28 at 11 58 58 am" src="https://user-images.githubusercontent.com/34310682/51824776-44042e00-22f4-11e9-966e-905118445e14.png">

`GET  /api/tags`
<img width="1211" alt="screen shot 2019-01-28 at 12 07 53 pm" src="https://user-images.githubusercontent.com/34310682/51825246-65b1e500-22f5-11e9-9e7d-357c2aa2268f.png">

`PATCH /api/article/<article-slug>`
<img width="1212" alt="screen shot 2019-01-28 at 11 53 58 am" src="https://user-images.githubusercontent.com/34310682/51824869-7f9ef800-22f4-11e9-9f55-becdc0a0c493.png">

<img width="1210" alt="screen shot 2019-01-28 at 11 54 12 am" src="https://user-images.githubusercontent.com/34310682/51824895-934a5e80-22f4-11e9-99d2-50428c915d0d.png">

#### Related story
[#162725267](https://www.pivotaltracker.com/story/show/162725267)

[![Run in Postman](https://run.pstmn.io/button.svg)](https://app.getpostman.com/run-collection/9a1fb51a44517a13f838)
